app/controllers/admin/loans_controller.rb

```diff
@@ -63,6 +73,12 @@ def destroy
     end
   end
 
+  def prepare_event(cal_event)
+    content = render_to_string(partial: "admin/calendar/event", locals: {cal_event: cal_event}).html_safe
+    cal_event[:title] = content
+    @events.push(cal_event)
```

Comment:
Fuzzy discussed that we would add an event model and event controller to handle events. I would prefer this not be duplicated, but thought this work would be handed off to someone with more expertise. Let's discuss.